configsuite/config.py

```diff
@@ -300,9 +318,9 @@ def _build_dict_snapshot(self, config, schema):
             ]
         )
 
-    def _build_snapshot(self, config, schema):
+    def _build_snapshot(self, config, schema, key_type=None):
         if config is None:
-            return None
+            return _RETURN_TYPE.get(key_type.name)
```

Comment:
This is an obvious miss if the `config is `None` and the `key_type` has default value